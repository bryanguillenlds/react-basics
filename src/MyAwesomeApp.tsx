import type { CSSProperties } from 'react';

const name = 'Bryan';
const lastName = 'Guillen';

const favoriteGames = [
  'The Witcher 3',
  'Elden Ring',
  'The Last of Us',
  'The Last of Us 2',
  'The Last of Us 3',
];

const styles: CSSProperties = {
  color: 'red',
};

export const MyAwesomeApp = () => {
  return (
    <>
      <h1 style={styles}> {name} </h1>
      <h3> {lastName} </h3>
      <ul>
        {favoriteGames.map((game) => (
          <li key={game}>{game}</li>
        ))}
      </ul>
    </>
  );
};
