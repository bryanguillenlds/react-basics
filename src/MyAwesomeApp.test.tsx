import { describe, it, expect } from 'vitest';
import { MyAwesomeApp } from './MyAwesomeApp';
import { render, screen } from '@testing-library/react';

describe('MyAwesomeApp', () => {
  it('should render first and last name correctly with container', () => {
    const { container } = render(<MyAwesomeApp />);

    const h1 = container.querySelector('h1');
    expect(h1?.textContent).toContain('Bryan');
  });

  it('should render first and last name correctly with screen', () => {
    render(<MyAwesomeApp />);

    const h1 = screen.getByRole('heading', { level: 1 });
    expect(h1.innerHTML).toContain('Bryan');
  });
});
