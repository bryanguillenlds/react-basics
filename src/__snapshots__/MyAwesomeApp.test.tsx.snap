// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match the snapshot 1`] = `
<div>
  <h1
    style="color: red;"
  >
     
    Bryan
     
  </h1>
  <h3>
     
    Guillen
     
  </h3>
  <ul>
    <li>
      The Witcher 3
    </li>
    <li>
      Elden Ring
    </li>
    <li>
      The Last of Us
    </li>
    <li>
      The Last of Us 2
    </li>
    <li>
      The Last of Us 3
    </li>
  </ul>
</div>
`;

exports[`MyAwesomeApp > should match the snapshot using screen 1`] = `
<div
  data-testid="my-awesome-app"
>
  <h1
    style="color: red;"
  >
     
    Bryan
     
  </h1>
  <h3>
     
    Guillen
     
  </h3>
  <ul>
    <li>
      The Witcher 3
    </li>
    <li>
      Elden Ring
    </li>
    <li>
      The Last of Us
    </li>
    <li>
      The Last of Us 2
    </li>
    <li>
      The Last of Us 3
    </li>
  </ul>
</div>
`;

exports[`MyAwesomeApp > should match the snapshot with render 1`] = `
<div>
  <div
    data-testid="my-awesome-app"
  >
    <h1
      style="color: red;"
    >
       
      Bryan
       
    </h1>
    <h3>
       
      Guillen
       
    </h3>
    <ul>
      <li>
        The Witcher 3
      </li>
      <li>
        Elden Ring
      </li>
      <li>
        The Last of Us
      </li>
      <li>
        The Last of Us 2
      </li>
      <li>
        The Last of Us 3
      </li>
    </ul>
  </div>
</div>
`;
